### PR TITLE
ROX-18223: version compatibility support exception handling

### PR DESCRIPTION
**Pull request by jschnath**
## Description

Add a support exception to compatibility tests for latest central with sensor 3.74.x. Add functionality to find the latest patch for a given version. 

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Checked CI logs and artifacts. The images below show that the support exception version is being tested as expected.
image
image

### CODE REVIEW

**Review by fredrb — APPROVED** [review]
Nice! Looks good to me. Maybe we should get a quick look at this from someone in infra as well just to be sure? @gavin-stackrox @tommartensen

File: scripts/ci/jobs/gke_version_compatibility_tests.py [hunk 1]
```diff
@@ -38,6 +41,11 @@
 # Latest sensor vs central versions in central_chart_versions
 test_tuples.extend([Chart_versions(central_version=central_chart_version, sensor_version=latest_tag) for central_chart_version in central_chart_versions])
 
+# Support exception for latest central and sensor 3.74 as per https://issues.redhat.com/browse/ROX-18223
+support_exceptions = [Chart_versions(central_version=latest_tag, sensor_version=get_latest_helm_chart_version_for_specific_release("stackrox-secured-cluster-services", Release(major=3, minor=74)))]
```
Comment (tommartensen) [inline scripts/ci/jobs/gke_version_compatibility_tests.py:45]: Each item in the support exceptions could be in a new line to improve readability..

File: scripts/ci/jobs/gke_version_compatibility_tests.py [hunk 2]
```diff
@@ -12,8 +12,11 @@
 from collections import namedtuple
 from compatibility_test import make_compatibility_test_runner
 from get_latest_helm_chart_versions import get_latest_helm_chart_versions
+from get_latest_helm_chart_versions import get_latest_helm_chart_version_for_specific_release
```
Comment (tommartensen) [inline scripts/ci/jobs/gke_version_compatibility_tests.py:15]:
```suggestion
from get_latest_helm_chart_versions import get_latest_helm_chart_versions,  get_latest_helm_chart_version_for_specific_release
```

File: .openshift-ci/get_latest_helm_chart_versions.py [hunk 3]
```diff
@@ -42,6 +42,9 @@ def main(argv):
     helm_versions = get_latest_helm_chart_versions("stackrox-secured-cluster-services", n)
     logging.info(f"Helm chart versions for the latest {n} releases:")
     print("\n".join(helm_versions))
+    specific_release=Release(major=3, minor=74)
```
Comment (tommartensen) [inline .openshift-ci/get_latest_helm_chart_versions.py:45]:
```suggestion
    specific_release = Release(major=3, minor=74)
```

Comment (tommartensen) [reply .openshift-ci/get_latest_helm_chart_versions.py:45]: Also this could be defined above the main function with a reference to the support exception.

**Review by tommartensen — COMMENTED** [review]
Just few minor things, no blockers from my side

File: scripts/ci/jobs/gke_version_compatibility_tests.py (hunk 1 above)
Comment (jschnath) [reply scripts/ci/jobs/gke_version_compatibility_tests.py:45]: Well so far there is only one support exception so no line breaks necessary yet

File: scripts/ci/jobs/gke_version_compatibility_tests.py (hunk 2 above)
Comment (jschnath) [reply scripts/ci/jobs/gke_version_compatibility_tests.py:15]: Done

File: .openshift-ci/get_latest_helm_chart_versions.py (hunk 3 above)
Comment (jschnath) [reply .openshift-ci/get_latest_helm_chart_versions.py:45]: Sure, I'll do that

File: .openshift-ci/get_latest_helm_chart_versions.py [hunk 4]
```diff
@@ -35,13 +35,19 @@
 # period.
 num_releases_default = 3
 
+# For support exceptions we may need to get the latest patch for a specific release that is not within the
+# last N versions. In that case get_latest_helm_chart_version_for_specific_release will provide the latest
+# patch of the input release.
+sample_support_exception Release(major=3, minor=74)
```
[the remark this answers is not in the document]
Comment (jschnath) [reply .openshift-ci/get_latest_helm_chart_versions.py]: Oops, fixed

**Pull request merged**